config/routes.rb

```diff
@@ -401,6 +401,15 @@
           put 'test_connection'
         end
         resources :images, :except => [:show]
+        resources :key_pairs, :only => [:index] do
+          member do
+            get 'download'
+          end
+          collection do
+            put 'recreate'
```

Comment:
this is not mapped to REST semantics, you are doing a PUT on a collection, imho it should be a PUT on the key itself.
it makes sense to do a POST on a collection (creates a new instance of a key) but not a PUT IMHO.